File: Visual Basic/PROJECTS/repos/Lotto/Lotto/Form1.vb
Public Class Form1
    Private Sub Clear()
        Label1.Text = "..."
        Label2.Text = "..."
        Label3.Text = "..."
        Label4.Text = "..."
        Label5.Text = "..."
        Label6.Text = "..."
        BtnPick.Enabled = True
        BtnReset.Enabled = False
    End Sub

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Clear()
        Randomize()
    End Sub

    Private Sub BtnReset_Click(sender As Object, e As EventArgs) Handles BtnReset.Click
        Clear()
    End Sub

    Private Sub BtnPick_Click(sender As Object, e As EventArgs) Handles BtnPick.Click

        ' Declare working variables.
        Dim i, r, temp, nums(60)

        ' Fill elements 1-59 with integers 1 to 59.
        For i = 1 To 59
            nums(i) = i
        Next

        ' Shuffle the values in elements 1-59.
        For i = 1 To 59
            r = Int(59 * Rnd() + 1)
            temp = nums(i)
            nums(i) = nums(r)
            nums(r) = temp
        Next

        ' Display the values in elements 1-6.
        Label1.Text = nums(1)
        Label2.Text = nums(2)
        Label3.Text = nums(3)
        Label4.Text = nums(4)
        Label5.Text = nums(5)
        Label6.Text = nums(6)

        ' Set the button states to Done.
        BtnPick.Enabled = False
        BtnReset.Enabled = True


    End Sub
End Class

File: Visual C#/SRC/Code/Lotto/Form1.cs
namespace Lotto
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Clear()
        {
            label1.Text = "...";
            label2.Text = "...";
            label3.Text = "...";
            label4.Text = "...";
            label5.Text = "...";
            label6.Text = "...";
            BtnPick.Enabled = true;
            BtnReset.Enabled = false;
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            Clear();
        }

        private void BtnPick_Click(object sender, EventArgs e)
        {
            // Create a random number generator object.  
            Random rnd = new Random();

            // Create an array to hold numbers 1-59.
            int[] seq = new int[59];

            // Fill array elements 0-58 with numbers 1-59.
            for (int i = 1; i < 60; i++) seq[i - 1] = i;

            // Shuffle the values in all array elements.
            for (int i = 0; i < 59; i++)
            {
                int j = (rnd.Next() % 59);
                int k = seq[i]; seq[i] = seq[j]; seq[j] = k;
            }

            // Display the values in array elements 1-6.
            label1.Text = seq[1].ToString();
            label2.Text = seq[2].ToString();
            label3.Text = seq[3].ToString();
            label4.Text = seq[4].ToString();
            label5.Text = seq[5].ToString();
            label6.Text = seq[6].ToString();

            // Set the Button state to Done.
            BtnPick.Enabled = false;
            BtnReset.Enabled = true;
        }

        private void BtnReset_Click(object sender, EventArgs e)
        {
            Clear();
        }
    }
}